cachito/workers/tasks/gomod.py

```diff
@@ -113,17 +115,26 @@ def fetch_gomod_source(request_id, dep_replacements=None, package_configs=None):
             env_vars.update(config.cachito_default_environment_variables.get("gomod", {}))
         else:
             env_vars = None
-        update_request_with_package(request_id, gomod["module"], env_vars, package_subpath=subpath)
-        update_request_with_deps(request_id, gomod["module"], gomod["module_deps"])
+
+        pkg_info = gomod["module"]
+
+        update_request_with_package(request_id, pkg_info, env_vars, package_subpath=subpath)
+        update_request_with_deps(request_id, pkg_info, gomod["module_deps"])
+        packages_json_data.add_package(pkg_info, subpath, gomod["module_deps"])
 
         # add package deps
         for package in gomod["packages"]:
-            package_subpath = _package_subpath(
-                gomod["module"]["name"], package["pkg"]["name"], subpath
-            )
-            update_request_with_package(request_id, package["pkg"], package_subpath=package_subpath)
+            pkg_info = package["pkg"]
+            package_name = pkg_info["name"]
+            package_subpath = _package_subpath(pkg_info["name"], package_name, subpath)
```

Comment:
```suggestion
            package_subpath = _package_subpath(mod_info["name"], pkg_info["name"], subpath)
```
See previous suggestion. Currently, the code has a bug where `pkg_info["name"]` is always the same as `package_name`